sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -81,9 +80,13 @@ export class ServiceBusClient {
     }
     this.fullyQualifiedNamespace = this._connectionContext.config.host;
     this._clientOptions.retryOptions = this._clientOptions.retryOptions || {};
-    this._clientOptions.retryOptions.timeoutInMs = getRetryAttemptTimeoutInMs(
-      this._clientOptions.retryOptions
-    );
+
+    let timeoutInMs = this._clientOptions.retryOptions.timeoutInMs;
+    timeoutInMs = timeoutInMs == undefined ? Constants.defaultOperationTimeoutInMs : timeoutInMs;
+    if (typeof timeoutInMs !== "number" || !isFinite(timeoutInMs) || timeoutInMs <= 0) {
+      throw new Error(`${timeoutInMs} is an invalid value for retryOptions.timeoutInMs`);
+    }
+    this._clientOptions.retryOptions.timeoutInMs = timeoutInMs;
```

Comment:
Do we **have** to set the timeout to the default value here? This forces us to use `!` every time we want to use `retryOptions.timeoutInMs` in all other places. 

Since the interface allows the timeout to not be set at all, I would suggest that we only do the error check here and copy over the `timeoutInMs == undefined ? Constants.defaultOperationTimeoutInMs : timeoutInMs;` to each place where the timeout is used.

Reply:
Updated messageSender with this, the other methods already have this. And thus avoided `!` at all the places.

Reply:
So, now that all consumers of retry options will check for undefined and fallback to the default value, do you need the above where you set `this._clientOptions.retryOptions.timeoutInMs`?

Reply:
Ah, if the timeout is undefined, `typeof timeoutInMs !== "number"` will be satisfied and an error is thrown. I thought it to be an overkill to allow undefined by placing more guards.

Reply:
Then add the `timeout != undefined` to your if check

```
if (timeout != undefined && (typeof timeoutInMs !== "number" || !isFinite(timeoutInMs) || timeoutInMs <= 0))
```

Reply:
Done!!